src/npctalk.cpp

```diff
@@ -2234,7 +2235,12 @@ void talk_effect_fun_t::set_message( const JsonObject &jo, const std::string &me
             }
         }
         if( popup_msg ) {
-            popup( translated_message, PF_NONE );
+            const auto new_win = [translated_message]() {
+                query_popup pop;
+                pop.message( "%s", translated_message );
+                return pop.get_window();
+            };
```

Comment:
I suggest create the window without hacking it using `query_popup`.

Reply:
I could copy paste the code but why duplicate it?

Reply:
Create a common function and use it in both `query_popup` and here then.

Reply:
After spending a while trying to do that, I ended up screwing up the original functionality around buttons rather than cleaning up the code.  I agree this is ugly but it works and avoids code duplication and refactoring the ui code hurts.